sensor/common/registry/registry_store.go

```diff
@@ -132,6 +132,8 @@ func (rs *Store) Cleanup() {
 	rs.cleanupClusterLocalRegistryHosts()
 	rs.cleanupDelegatedRegistryConfig()
 	rs.tlsCheckCache.Cleanup()
+
+	metrics.ResetRegistryMetrics()
 }
 
 func (rs *Store) cleanupRegistries() {
```

Comment:
> Will this mess with the metrics?

Yes intentionally - the metrics reset in`metrics.ResetRegistryMetrics()` are meant to represent 'current state', since the cleanup method zeros out the contents of the registry store, the associated metrics should be zeroed out as well.

> when is this called? 

It's called on sensor shutdown and I believe when Sensor re-establishes a connection to Central, below is the reverse call flow

https://github.com/stackrox/stackrox/blob/a26fd772cb732968e2672b3209b9753509639b5d/sensor/kubernetes/listener/resources/store_provider.go#L84-L88

https://github.com/stackrox/stackrox/blob/e8a3a75590d7c33236fbf7242abb4470cf149578/sensor/kubernetes/listener/listener_impl.go#L85-L91

Sensor Shutdown:
https://github.com/stackrox/stackrox/blob/e8a3a75590d7c33236fbf7242abb4470cf149578/sensor/kubernetes/eventpipeline/pipeline_impl.go#L116

Sensor Reconnect:
https://github.com/stackrox/stackrox/blob/e8a3a75590d7c33236fbf7242abb4470cf149578/sensor/kubernetes/eventpipeline/pipeline_impl.go#L131